modules/core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/impl/ExportedDocumentImpl.java

```diff
@@ -310,7 +310,7 @@ protected void readProperty(Element parent, Namespace targetNs, Field field, Obj
 
     protected final void readBlob(Element element, ComplexType ctype, Blob blob, boolean inlineBlobs)
             throws IOException {
-        String blobPath = Integer.toHexString(RANDOM.nextInt()) + ".blob";
+        String blobPath = blob.getFilename() != null ? blob.getFilename() : Integer.toHexString(RANDOM.nextInt()) + ".blob";
```

Comment:
This is problematic because if a document has several attached blobs with the same filename, you will have a collision. So you would need to check in the `blobs` map (when not using `inlineBlobs`) and have a way to choose an alternate name that is not already in the map.

Reply:
> Thanks for your contribution. I made a comment for potential improvement.
> 
> NXP-28892 is about retention, is the right ticket?
> 
> I trigger a T&P to validate the change: https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-kleturc-3/32/

I apologize, the correct ticket is 28897